llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -560,6 +560,96 @@ def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack", [
   }];
 }
 
+def IREELinalgExt_UnPackOp : IREELinalgExt_Op<"unpack", [
+  DeclareOpInterfaceMethods<ReifyRankedShapedTypeOpInterface>,
+  DeclareOpInterfaceMethods<LinalgExtInterface>,
+  DeclareOpInterfaceMethods<TilingInterface,
+    ["getIterationDomain",
+     "getLoopIteratorTypes",
+     "generateScalarImplementation"]>,
+  DeclareOpInterfaceMethods<MemoryEffectsOpInterface>
+]>{
+  let summary = "unpack operation";  
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+    Variadic<AnyShaped>:$outputs,
+    DefaultValuedAttr<I64ArrayAttr, "{}">:$dims_pos,
+    Variadic<Index>:$inner_tiles,
+    I64ArrayAttr:$static_inner_tiles,
+    Optional<AnyType>:$padding_value);
+
+  let results = (outs Variadic<AnyRankedTensor>:$results);
+  let assemblyFormat = [{
+    attr-dict
+    $inputs
+    (`padding_value` `(` $padding_value^ `:` type($padding_value) `)`)? 
```

Comment:
We don't need padding_value for unpack right? The unpack op is `extract_slice + relayout + transpose variants` to me.

If the goal is to share more common snippets between pack and unpack in `generatePackOpOrUnPackScalarImplementationBody`, we can pass a `packPaddingValue` to the method. The unpack op can pass nullptr to `generatePackOpOrUnPackScalarImplementationBody` in this case.

Reply:
We need the padding value to know what `slice` to extract out. Basically have all the information to apply packing in reverse?

Reply:
I thought what slice to extract out can be derived from input shapes, inner_tiles, and dim_pos? When computing the shape, padding value is not needed.

See an example from matmul -> mmt4d below. The input of unpack op is a 4D tensor. What we've done:

1. collapse input into a 2D tensor. 
2. extract a top-left slice from 2D tensor.

There are no padding values involved.

https://github.com/iree-org/iree/blob/0bca135fc03950c1b5e777f0f8127f97be41da8e/compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp#L294-L305